lsr-role2collection.py

```diff
@@ -0,0 +1,772 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# (c) 2020 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+
+#     (see https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Usage:
+# lsr-role2collection.py [--namespace NAMESPACE]
+#                        [--collection COLLECTION]
+#                        --src-path SRC_PATH
+#                        --dest-path DEST_PATH
+#                        --role ROLE | --molecule
+#                        (if --molecule is set, SRC_PATH/template is expected)
+#                        [--replace-dot STR]
+#                        [-h]
+
+import argparse
+import os
+import re
+import fnmatch
+from shutil import copytree, copy2, ignore_patterns, rmtree
+
+from pathlib import Path
+
+ROLE_DIRS = (
+    'defaults',
+    'files',
+    'handlers',
+    'meta',
+    'tasks',
+    'templates',
+    'vars',
+)
+
+PLUGINS = (
+    'action_plugins',
+    'become_plugins',
+    'cache_plugins',
+    'callback_plugins',
+    'cliconf_plugins',
+    'connection_plugins',
+    'doc_fragments',
+    'filter_plugins',
+    'httpapi_plugins',
+    'inventory_plugins',
+    'library',
+    'lookup_plugins',
+    'module_utils',
+    'netconf_plugins',
+    'shell_plugins',
+    'strategy_plugins',
+    'terminal_plugins',
+    'test_plugins',
+    'vars_plugins'
+)
+
+TESTS = (
+    'tests',
+)
+
+DOCS = (
+    'docs',
+    'design_docs',
+    'examples',
+    'README.md',
+    'DCO',
+)
+
+MOLECULE = (
+    '.ansible-lint',
+    'custom_requirements.txt',
+    'molecule',
+    'molecule_extra_requirements.txt',
+    'tox.ini',
+    '.travis',
+    '.travis.yml',
+    '.yamllint_defaults.yml',
+    '.yamllint.yml',
+)
+
+DO_NOT_COPY = (
+    '.github',
+    '.gitignore',
+    '.lgtm.yml',
+    '.tox',
+    '.venv',
+    'ansible_pytest_extra_requirements.txt',
+    'artifacts',
+    'pylint_extra_requirements.txt',
+    'pylintrc',
+    'pytest_extra_requirements.txt',
+    'run_pylint.py',
+    'scripts',
+    'semaphore',
+    'standard-inventory-qcow2',
+    'tuned_requirements.txt',
+)
```

Comment:
Many of them will go away when we move the shared CI files into a shared git submodule, right?

Reply:
most of them, yes